src/utils/api.js

```diff
@@ -20,14 +39,49 @@ let newListing = {
   condition: "",
   location: "",
   tradeRequired: false,
-  user_id: ""
+  user_id: "",
 };
 
+let listingRef;
 export const postListing = async (event, newListing) => {
   event.preventDefault();
-
   const docRef = await addDoc(collection(db, "listings"), newListing);
   console.log("Document written to Listings with ID: ", docRef.id);
+  listingRef = docRef.id;
+  uploadImage(event);
+  return listingRef;
+};
+
+const storageRef = ref_storage(storage);
+const metadata = {
```

Comment:
same with this, can it just be defined within the function it's needed? Doesn't appear to have any out of scope dependencies